src/main/java/com/google/sps/servlets/AutoEventProcessor.java

```diff
@@ -0,0 +1,318 @@
+        return input.replaceAll("\"", "");
+    }
+
+    @Override
+    public void doGet(HttpServletRequest req, HttpServletResponse res) {
+    }
+
+    @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse res) throws IOException {
+
+        res.setContentType("application/json;");
+
+        // Users get sent as a JsonArray of Strings, use Gson Library to load them into
+        // a string array
+        String[] users = new Gson().fromJson(req.getParameter("userList"), String[].class);
+
+        /* Scheduling Constraints */
+        int startConstraint = Integer.parseInt(req.getParameter("startRes"));
+        int endConstraint = Integer.parseInt(req.getParameter("endRes"));
+        int meetingLength = Integer.parseInt(req.getParameter("meetingLength"));
+        
+        try {
+
+            List<Calendar> calendars = getCalendars(users);
+
+            int[][] freeBusyTimes = getFreeBusyTimes(calendars);
+            /* 
+                 Below is for Testing
+
+            for(int hour = 0; hour < 24; hour++){
+                for(int day = 0; day < 7; day++){
+                    System.out.print(freeBusyTimes[day][hour] + " ");
+                }
+                System.out.println();
+            }
+            */
+            String eventTimesJson = processFreeBusyTimes(freeBusyTimes, startConstraint, endConstraint, meetingLength);
+
+            res.getWriter().println(eventTimesJson);
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            System.out.println("Error Occured");
+            res.getWriter().println("{\"error\": true}");
+        }
+
+        doGet(req, res);
+
+    }
+
+    // ====================================Get Calendars Start ========================================//
+    private static List<Calendar> getCalendars(String[] users) throws org.apache.http.ParseException, IOException, GeneralSecurityException {
+
+        List<Calendar> calendars = new ArrayList<Calendar>();
+        final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+        Credential creds;
+        TokenResponse tokenResponse = new TokenResponse();
+
+        for (String user : users) {
+
+            String access_token = DatastoreModule.getUserAccessTokenWithEmail(user);
+
+            if (access_token == null || access_token.equals("")) {
+                System.out.println("User: " + user + "\nIs not Registered");
+            } else {
+                tokenResponse.setAccessToken(access_token);
+                creds = buildCreds(HTTP_TRANSPORT, tokenResponse);
+                try {
+                    Calendar service = new Calendar.Builder(HTTP_TRANSPORT, JSON_FACTORY, creds)
+                            .setApplicationName(APPLICATION_NAME).build();
+
+                    calendars.add(service);
+
+                } catch (Exception e) {
+                    // This may cause a loop
+                    return getCalendars(users);
```

Comment:
Great comment. Is there a better way to make sure you don't run into an infinite loop here? 

One janky way of doing it would be to add a class level variable that keeps track how many retries and after a certain amount it could just return an error and reset its value.